### PR TITLE
T23: Voting Contest

### PR DESCRIPTION
**Pull request by Nandhini-Sakthi**

### CODE REVIEW

**Comment by tmmvigneshm** [comment]
Good.

File: Program.cs [hunk 1]
```diff
@@ -1,4 +1,51 @@
...
+
+namespace Training {
+   #region Program ------------------------------------------------------------------------------
+   /// <summary>Voting Contest</summary>
+   internal class Program {
+      #region Methods ---------------------------------------------
+      /// <summary>getting input from the user</summary>
+      static void Main () {
+         Console.WriteLine ("Enter the string: ");
+         string input = Console.ReadLine ()!;
```
Comment (tmmvigneshm) [inline Program.cs:21]: Fix nullable string warning.

File: Program.cs [hunk 2]
```diff
@@ -1,4 +1,51 @@
...
+         FindWinner (input, out char winner, out int votes);
+         Console.WriteLine ($"{winner}, {votes} => Winner and votes");
+      }
+      /// <summary>find the winner based on votes</summary>
+      /// <param name="input">word</param>
+      /// <param name="winner">letter with maximum votes</param>
+      /// <param name="votes">votes</param>
+      static void FindWinner (string input, out char winner, out int votes) {
```
Comment (tmmvigneshm) [inline Program.cs:29]: Move this method outside of Main method.

File: Program.cs [hunk 3]
```diff
@@ -1,4 +1,51 @@
...
+         Dictionary<char, int> voteCount = new Dictionary<char, int> ();//store character and votes.
+         foreach (char c in input.ToLower ()) {
+            if (Char.IsLetter (c)) {
+               if (voteCount.ContainsKey (c)) voteCount[c]++;
+               else voteCount[c] = 1;
+            }
```
Comment (tmmvigneshm) [inline Program.cs:35]: Remove curly braces for single line statement.

File: Program.cs [hunk 4]
```diff
@@ -1,4 +1,51 @@
...
+         char currentWinner = ' ';
+         int maxVotes = 0;
+         foreach (var kvp in voteCount) {
+            if (kvp.Value > maxVotes) {//find which char has the most votes.
+               currentWinner = kvp.Key;
+               maxVotes = kvp.Value;
+            }
+         }
+         winner = currentWinner;
+         votes = maxVotes;
```
Comment (tmmvigneshm) [inline Program.cs]: Whole logic can be simplified by using Dictionary.OrderbyDescending method.
Please study about Dictionary.OrderByDescending, and you should use the dictionary value for descending.

e.g.  voteCount.OrderByDesc......(......);

File: Program.cs [hunk 5]
```diff
@@ -1,4 +1,46 @@
...
+         FindWinner (input, out char winner, out int votes);
+         Console.WriteLine ($"{winner}, {votes} => Winner and votes");
+      }
+      /// <summary>find the winner based on votes</summary>
+      /// <param name="input">word</param>
+      /// <param name="winner">letter with maximum votes</param>
+      /// <param name="votes">votes</param>
+      static void FindWinner (string input, out char winner, out int votes) {
+         // Store character and votes.
+         Dictionary<char, int> voteCount = new Dictionary<char, int> ();
```
Comment (tmmvigneshm) [inline Program.cs]: Simplify the assignment with just new ();

File: Program.cs [hunk 6]
```diff
@@ -1,4 +1,46 @@
...
+         foreach (char c in input.ToLower ()) {
+            if (Char.IsLetter (c)) {
+               if (voteCount.ContainsKey (c)) voteCount[c]++;
+               else voteCount[c] = 1;
```
Comment (tmmvigneshm) [inline Program.cs]: Can you try rewriting this logic with ternary operator ?
Just try and see.

Comment (Nandhini-Sakthi) [reply Program.cs]: All the changes are made in the latest commit.

**Comment by tmmvigneshm** [comment]
Good.

**Pull request closed without merging**